lib/src/kvs/tests/mod.rs

```diff
@@ -257,7 +266,19 @@ mod fdb {
 	}
 
 	async fn clear_cluster(mut tx: Transaction) -> Result<(), Error> {
-		tx.delp(vec![], u32::MAX).await?;
+		if let Err(err) = tx.delp(vec![], u32::MAX).await {
+			let second_err = tx.cancel().await;
+			return match second_err {
+				Ok(_) => Err(err),
+				Err(e2) => {
+					error!(
+						"Failed to cancel transaction: {}, original error cause was: {}",
+						e2, err
+					);
```

Comment:
I understand the need to log an error at source like this but both logging and returning an error leads to the error being logged more than once. A good practice is to either log an error or return it, not both.

If the motive for doing this is to create a more descriptive error message, you can add a variant dedicated to this so you can re-use the message you used here. If the motive was to make it easier to track down where the error occurred, that's typically solved by adding backtrace support in error types. Something we don't currently do, unfortunately, but a descriptive error message can help ease that somewhat.

Reply:
Resolved